### ISSUE TITLE
Feature Request: Status while importing

### ISSUE THREAD

**Issue by trashcoder** [xemle/home-gallery#15]
It would be nice to have a status in percent or something like 1432 of 38474 Pictures imported.

**Comment by xemle** [comment]
TL;DR the current non-progress state is on purpose. The recommended import is the initial (progressive) import via `./gallery run import --initial`.

Looong version: I am totally aware of the awesome progress information on long taking processes like importing a gallery data. My assumption is that a user imports 50k - 100k media files (0,5 TB - 1 TB data) with some videos and wants to see as early as possible some pic on the gallery.

Due the architecture of HomeGallery the files need to be indexed and the file checksum needs to be calculated which are used as unique file ids. Afterwards the preview files and web compatible video previews need to be calculated. As final step the database is build. So there is no intermediate step to shortcut to present pictures in the browser.

If we process the whole media archive at once it is time consuming for checksums, image and video previews. As mentioned in the docs, the image previews will take days, the video previews will take weeks depending on your amount of pictures, videos and host machine.

Since the assumed standard user wants to see early results, the recommended import of an archive should be incremental (or by chunks) via `./gallery run import --initial`. Than the files are indexed and processed in chunks and the database is build successive. The browser will reload automatically if the database changes. So the user sees some pictures early while the whole archive import is in progress for weeks.

The internals of this progress is done via the indexer and the `--add-limits` parameter listed by `./gallery index -h` (see also the [source comments](https://github.com/xemle/home-gallery/blob/master/packages/index/src/limit-filter.js). This uses the gallery characteristic that the gallery knows only the files, which are indexed. So the indexer stops after adding some new files (afterwards these new files are processed, the database is updated, new files are indexed and so on and so forth).

So the gallery can not be aware of all files. Only of the current chunk - this could be fixed. Yes. Or the add limit could be limit to the file checksum calculation to have at least the total file count since the checksum is required to process the files... Overall I am impatient as well and think the current solution could be a good balance between waiting weeks and see some progress. You might inspect also the index stats via `./gallery index stats -i <index files>` how many files are in the index.

Reading this internals and the background: What solution/improvement would you suggest/propose fitting the given context?

**Comment by xemle** [comment]
Hi. I found a way how to support progress information on the import.

Here my pseudo code

```
stats = collect overall file stats (omit checksum/id calculation)
for each chunk:
  index chunk honor stats
  extract chunk honor stats
  build database honor stats
end
```

So we walk only once over all file in the `run import` command and pass these stat information towards each chunk. Each chunk step of index, extract and database could create a progress information and an estimate of required time. I assume that the stat information wont change a lot while running the import. Would that work for you?

I am very happy to find a solution draft. This feature request needs some refactoring and work. Currently I can not give an estimation, when this feature will be implemented. If someone likes to implement in the meanwhile I could give some hints and support.

Thank you again for your input and your request.

**Comment by trashcoder** [comment]
Yes, that would be great! Really cool! 😀

**Comment by xemle** [comment]
I am closing this ticket due inactivity. Please do not hesitate to reopen it
